### PR TITLE
fix breaking rule

### PR DESCRIPTION
**Pull request by pancaspe87**
This resolves https://github.com/atlassian-labs/compiled/issues/1312

### CODE REVIEW

File: packages/eslint-plugin/src/utils/create-no-tagged-template-expression-rule/index.ts [hunk 1]
```diff
@@ -39,11 +39,18 @@ export const createNoTaggedTemplateExpressionRule =
             return;
           }
 
+          const args = toArguments(source, quasi);
+
+          // Skip invalid CSS
+          if (args.length < 1) {
```
Comment (pancaspe87) [inline packages/eslint-plugin/src/utils/create-no-tagged-template-expression-rule/index.ts:45]: RFC: This is to prevent eslint rule to break when dealing with invalid CSS such as `margin-top 0;` ( missing colon symbol).
However, maybe we should throw an error instead.

File: packages/eslint-plugin/src/utils/create-no-tagged-template-expression-rule/to-arguments.ts [hunk 2]
```diff
@@ -130,7 +135,10 @@ export const toArguments = (source: SourceCode, template: ESTree.TemplateLiteral
   };
 
   for (const [i, quasi] of template.quasis.entries()) {
-    for (const char of quasi.value.raw) {
+    // Deal with multi selectors
+    const styleTemplateElement = quasi.value.raw.replace(/\n/g, '').replace(/\s+/g, ' ');
```
[the remark this answers is not in the document]
Comment (JakeLane) [reply packages/eslint-plugin/src/utils/create-no-tagged-template-expression-rule/to-arguments.ts:139]:
```suggestion
    const styleTemplateElement = quasi.value.raw.replace(/(\r\n|\n|\r)/gm, ' ').replace(/\s+/g, ' ');
```
In this case we should add a space, consider the following edge case:
```
div
h1 { ... }
```

Which will turn into:
```
divh1 { ... }
```

File: packages/eslint-plugin/src/utils/create-no-tagged-template-expression-rule/index.ts (hunk 1 above)
Comment (JakeLane) [reply packages/eslint-plugin/src/utils/create-no-tagged-template-expression-rule/index.ts:45]: Throwing an error will cause the lint rule to fail. This shouldn't be an error for the lint rule as it's expected to see invalid CSS in the AST. 

We should skip auto-fixing if we would end up with invalid code if transformed.

**Review by JakeLane — CHANGES_REQUESTED** [review]

File: packages/eslint-plugin/src/rules/no-styled-tagged-template-expression/__tests__/rule.test.ts [hunk 3]
```diff
@@ -899,5 +899,84 @@ tester.run('no-styled-tagged-template-expression', noStyledTaggedTemplateExpress
...
+      `,
+      output: `
+        import { styled } from '@compiled/react';
+
+        styled.div\`
+          color blue;
+        \`;
+      `,
+    },
   ]),
```
Comment (JakeLane) [inline packages/eslint-plugin/src/rules/no-styled-tagged-template-expression/__tests__/rule.test.ts:981]: Can you please also add a test for https://github.com/atlassian-labs/compiled/pull/1313#discussion_r1027385229

File: packages/eslint-plugin/src/utils/create-no-tagged-template-expression-rule/to-arguments.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (pancaspe87) [reply packages/eslint-plugin/src/utils/create-no-tagged-template-expression-rule/to-arguments.ts:139]: both good point!

I'll add test cases for the edge case.

File: packages/eslint-plugin/src/utils/create-no-tagged-template-expression-rule/index.ts (hunk 1 above)
Comment (pancaspe87) [reply packages/eslint-plugin/src/utils/create-no-tagged-template-expression-rule/index.ts:45]: absolutely right re auto-fixing!

**Review by liamqma — APPROVED** [review]

**Comment by pancaspe87** [comment]
Hey @liamqma re https://github.com/atlassian-labs/compiled/pull/1313#discussion_r1026077290

IIRC `return`; will prevent the autofixer to run. Hence, a `tagged-template-expression` won't be auto-fixed on save. `throw` will cause the lint rule to show the error only when running the command and may attempt to fix.

This is what @JakeLane meant in https://github.com/atlassian-labs/compiled/pull/1313#discussion_r1027385867

**Review by JakeLane — APPROVED** [review]

**Review by liamqma — APPROVED** [review]
